### ISSUE TITLE
nf_elem printing broken on Windows master

### ISSUE THREAD

**Issue by thofma** [Nemocas/Nemo.jl#859]
There is an easy work around, but it would be still good to figure out why this is failing. It is failing in the line
```julia
   cstr = ccall((:nf_elem_get_str_pretty, libantic), Ptr{UInt8},
                (Ref{nf_elem}, Ptr{UInt8}, Ref{AnticNumberField}),
                 x, string(var(parent(x))), parent(x))
```
The manual suggests to use `Cstring` instead of `Ptr{UInt8}` if the strings are null-terminated. I tried it with `Cstring` instead and also with `Ptr{Cchar}`, but to no avail. We probably need a Windows box to reproduce this locally.

**Comment by wbhart** [comment]
Don't we have other functions where we get a string? Those must work on Windows, right?

**Comment by thofma** [comment]
Yes it is strange. For example printing of `fq_elem` works, which is also done by calling into flint.

**Comment by wbhart** [comment]
Should we just try doing it the same way as we do fq_elem then, i.e. with unsafe_string and flint_free?

**Comment by thofma** [comment]
We are already doing it. I just posted the function that is failing. Afterwards there is an `unsafe_string(cstr)`, but he does not even get there.

**Comment by wbhart** [comment]
Oh, which UInt8 is causing the problem? The output or the variable?

**Comment by thofma** [comment]
I don't know. I don't have a Windows box and using CI I could trace it down only to this line. Here is my latest test: https://travis-ci.org/github/Nemocas/Nemo.jl/jobs/712990045 with this version:

```julia
function show(io::IO, x::nf_elem)
  println("1")
  println(parent(x))
  println(parent(parent(x).pol)(x))
  println(string(var(parent(x))))
  cstr = ccall((:nf_elem_get_str_pretty, libantic), Ptr{Cchar},
               (Ref{nf_elem}, Ptr{Cchar}, Ref{AnticNumberField}),
                 x, string(var(parent(x))), parent(x))
  println("2")
   s = unsafe_string(cstr)
  println("3")
  ccall((:flint_free, libflint), Nothing, (Ptr{Cchar},), cstr)
  println("4")
   s = replace(s, "/" => "//")
  println("5")
  print(io, s)
end```

**Comment by wbhart** [comment]
Ah, I see. I have a Windows box, but it is dangerously low on hard drive space. I keep cleaning it up, but Microsoft keep installing more and more stuff on there. I'll be getting a new box later this year which will have a bigger hard drive and I'll do a Windows source build on there. That should help tracking down stuff like that too.

Microsoft are also apparently fixing my editor this month, which has been broken for about 8 months!

**Comment by thofma** [comment]
Congratulations on the editor/console! I might be asking Dan to bring his Windows box next week.

**Comment by wbhart** [comment]
It's fast!

**Comment by thofma** [comment]
Dan helped me track down the bug. It is a problem with the binaries. We are working on a fix.

**Comment by thofma** [comment]
https://github.com/JuliaPackaging/Yggdrasil/issues/1436

I had a chat with the binary people. They are thinking about a proper fix.

**Comment by thofma** [comment]
Fixed with FLINT_jll >= 2.6.3.
